plugins/ccgx/fu-ccgx-hpi-device.c

```diff
@@ -292,11 +230,27 @@ fu_ccgx_hpi_device_i2c_read (FuCcgxHpiDevice *self,
 		g_prefix_error (error, "i2c read error: bulk xfer: ");
 		return FALSE;
 	}
-
 	/* 10 msec delay */
 	g_usleep (I2C_READ_WRITE_DELAY_US);
-	if (!fu_ccgx_hpi_device_wait_for_notify (self, NULL, error)) {
-		g_prefix_error (error, "i2c read error: ");
+	if (!g_usb_device_interrupt_transfer (fu_usb_device_get_dev (FU_USB_DEVICE (self)),
+					      self->ep_intr_in,
+					      i2c_status, sizeof(i2c_status), NULL,
+					      FU_CCGX_HPI_WAIT_TIMEOUT,
+					      NULL, error)) {
+		g_prefix_error (error, "i2c read error: intr xfer: ");
+		return FALSE;
+	}
+	
+	if (i2c_status[0] & CY_I2C_ERROR_BIT) {
+		if (!fu_ccgx_hpi_device_reset_i2c (self, CY_I2C_MODE_READ, error)) {
+			g_prefix_error (error, "i2c reset error: ");
+			return FALSE;
+		}
```

Comment:
i've said it before, don't do the reset in the error path...

Reply:
I can't understand the meaning of "don't do the reset in the error path"
Could you explain more?

Reply:
Okay, so valid bug you found, but I think the fix is better like the last patch in https://github.com/fwupd/fwupd/pull/1959